src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php

```diff
@@ -41,8 +41,8 @@ public function process(ContainerBuilder $container)
         try {
             $this->container = $container;
             foreach ($container->getDefinitions() as $id => $definition) {
-                if ($definition->isAutowired()) {
-                    $this->completeDefinition($id, $definition);
+                if (false !== $autowired = $definition->getAutowired()) {
```

Comment:
What about keep using `$definition->isAutowired()` here (as it stay valid), remove the newly added `$autowired` argument from `completeDefinition()` and use `$autowired = $defintion->getAutowired();` directly into?